crates/router/src/workflows/payment_method_mandate_details_revoke.rs

```diff
@@ -0,0 +1,136 @@
+            )
+            .await?;
+
+        let retry_count = process.retry_count;
+        let merchant_account = db
+            .find_merchant_account_by_merchant_id(&tracking_data.merchant_id, &key_store)
+            .await?;
+
+        let connector_name = tracking_data.connector.to_string();
+        let merchant_connector_account = helpers::get_merchant_connector_account(
+            state,
+            &tracking_data.merchant_id,
+            None,
+            &key_store,
+            &tracking_data.profile_id,
+            &connector_name,
+            Some(&tracking_data.merchant_connector_id),
+        )
+        .await?;
+        let connector_data = ConnectorData::get_connector_by_name(
+            &state.conf.connectors,
+            &connector_name,
+            GetToken::Connector,
+            Some(tracking_data.merchant_connector_id.clone()),
+        )?;
+        let connector_integration: services::BoxedMandateRevokeConnectorIntegrationInterface<
+            types::api::MandateRevoke,
+            types::MandateRevokeRequestData,
+            types::MandateRevokeResponseData,
+        > = connector_data.connector.get_connector_integration();
+
+        let router_data = utils::construct_mandate_revoke_router_data(
+            merchant_connector_account,
+            &merchant_account,
+            tracking_data.customer_id.clone(),
+            connector_name,
+            Some(tracking_data.connector_mandate_id),
+            None,
+        )
+        .await?;
+
+        let response = services::execute_connector_processing_step(
+            state,
+            connector_integration,
+            &router_data,
+            CallConnectorAction::Trigger,
+            None,
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+        match response.response {
+            Ok(_mandate) => {
+                db.as_scheduler()
+                    .finish_process_with_business_status(process, "COMPLETED_BY_PT")
+                    .await?;
+            }
+            Err(err) => {
+                // if not implemented end the task in the PT
+                if err.code == "IR_00" {
+                    db.as_scheduler()
+                        .finish_process_with_business_status(
+                            process,
+                            "NOT IMPLEMENTED FOR THE CONNECTOR",
+                        )
+                        .await?;
+                } else {
+                    // if connector err re-schedule task
+                    let mapping = process_data::PaymentMethodMandateRevokePTMapping::default();
+                    let time_delta = if retry_count == 0 {
+                        Some(mapping.default.start_after)
+                    } else {
+                        pt_utils::get_delay(retry_count + 1, &mapping.default.frequencies)
+                    };
+                    let schedule_time = pt_utils::get_time_from_delta(time_delta);
```

Comment:
Can we extract this logic out to a function? And shouldn't we be considering the connector-specific overrides instead of always going with the default?